File: src/171ExcelSheetColumnNumber/171ExcelSheetColumnNumber.go
package main

import "fmt"

func titleToNumber(s string) int {
	sum := 0
	for i := 0; i < len(s); i++ {
		sum = sum*26 + int(s[i]-'A') + 1
	}
	return sum
}

func main() {
	fmt.Printf("%v\n", titleToNumber("AB"))
}

File: src/171ExcelSheetColumnNumber/171ExcelSheetColumnNumber.rs
impl Solution {
    pub fn title_to_number(s: String) -> i32 {
        let mut iter = s.chars();
        let mut num = 0;
        while let Some(c) = iter.next() {
            num = num*26 + (c as u8 - 'A' as u8) as i32 + 1;
        }
        num
    }
}
